csrc/ir/nodes.cpp

```diff
@@ -2063,6 +2063,45 @@ void MmaOp::setMacro(MmaMacro macro) {
   attribute<MmaMacro>(ATTR_POS_MACRO) = macro;
 }
 
+std::vector<PolymorphicValue> MmaOp::evaluate(
+    const ExpressionEvaluator& ee,
+    const std::vector<PolymorphicValue>& inputs) const {
+  
+  const auto tv_a = inA()->as<TensorView>();
+  const auto tv_b = inB()->as<TensorView>();
+  NVF_CHECK(
+    tv_a->nDims() == tv_b->nDims(), 
+    "Either both or none of A and B should be batch");
+  // Verify that the broadcasted size is 3. 
+  NVF_CHECK(
+    tv_a->nDims() == 3,
+    "MmaOp::evaluate is not implemented for size: ", tv_a->nDims());
+
+  // Assumptions:
+  //    Currently, the evaluate method assumes that the MmaOp is preceded by a
+  //    broadcast. The inputs to MmaOp are broadcasted as the last dim for the
```

Comment:
Do we care about the MmaOp being preceded by a BroadcastOp? Is the `isBroadcast` check sufficient?

Reply:
I added the first check to verify that these broadcasts resulted from the `broadcastOp` preceding `fusedMatmulSum`. It is possible for the inputs to have been broadcasted at some other point and not for matmul.

Reply:
It is mainly to guard against this change so we are clear on what we expect in the `evaluate` method.

Reply:
> In that case, we may be removing dimensions which just happen to be broadcasted but by some other op.

You are probably referring to #1628, for which one of the solutions (probably the POR) is to have `MatmulScheduler` reject the preceding `BroadcastOp`s and therefore insert `Set` for tmaLoad between `BroadcastOp` and `MmaOp`. In that case, your code as is will work just fine **except** this `BroadcastOp` check :) That's why I think removing this check is better. 

Anyhow, leaving the check as is should be fine for now. I was mainly verifying the motivation behind it.